trin-core/src/portalnet/overlay_service.rs

```diff
@@ -280,6 +282,11 @@ impl<TContentKey: OverlayContentKey + Send> OverlayService<TContentKey> {
 
         let (response_tx, response_rx) = mpsc::unbounded_channel();
 
+        let metrics: Option<Metrics> = match enable_metrics {
```

Comment:
I think this can be replaces with [then](https://doc.rust-lang.org/std/primitive.bool.html#method.then):

```rust
        let metrics: Option<Metrics> = enable_metrics.then(|| Metrics::init(&protocol));
```